src/scalars/time/GraphQLYear.js

```diff
@@ -24,14 +25,15 @@ export default new GraphQLScalarType({
 
     serialize(value) {
         if (value instanceof Date) {
-            return getYearAsStr(value);
+            return getYearAsStr(value.getUTCFullYear().toString());
         } else if (typeof value === 'string' || value instanceof String) {
             if (!isNaN(Date.parse(value))) {
-                return getYearAsStr(new Date(value));
+                // when the date is in ISO format we need to only the year
+                return getYearAsStr(value.match(YEAR_REGEX)[0] || value);
             }
 
             validateYear(value);
-            return value.toString();
+            return getYearAsStr(value.match(YEAR_REGEX)[0] || value);
```

Comment:
I don't get this. Shouldn't you return just `value` here?

Reply:
I'll check, maybe I copy/pasted it.

Reply:
I've checked, it handles the cases where the year is negative value and it is not in format that could be parsed with standard JS Date type. There is a test case for it - `Wed, 10 Jun -2020 06:54:06 GMT` -> `-2020` . Altho, I am not sure that such date format is valid. I think UTC does not allow negative values, but there were some weird tests in the Niki's suite.